--- a/soplex/CMakeLists.txt
+++ b/soplex/CMakeLists.txt
@@ -43,6 +43,7 @@
 option(SANITIZE "should sanitizers be enabled if available" OFF)
 option(COVERAGE "enable coverage support" OFF)
 option(PAPILO "should papilo library be linked" ON)
+option(SOPLEX_BUILD_TESTS "Build the SoPlex test targets" ON)
 
 SET(COVERAGE_CTEST_ARGS "" CACHE STRING "additional ctest arguments for coverage")
 
@@ -251,7 +252,8 @@
 configure_file(${PROJECT_SOURCE_DIR}/soplex-config.cmake.in "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/soplex-config.cmake" @ONLY)
 
 add_subdirectory(src)
-add_subdirectory(tests/c_interface)
-add_subdirectory(check)
-
-enable_testing()
+if(SOPLEX_BUILD_TESTS)
+    add_subdirectory(tests/c_interface)
+    add_subdirectory(check)
+    enable_testing()
+endif()
